tests/functional/WebDriverTestCase.php

```diff
@@ -67,32 +67,30 @@ protected function setUp(): void
                 $firefoxOptions = new FirefoxOptions();
                 $firefoxOptions->addArguments(['-headless']);
                 $this->desiredCapabilities->setCapability(FirefoxOptions::CAPABILITY, $firefoxOptions);
+            } elseif ($browserName === WebDriverBrowserType::SAFARI) {
+                $this->serverUrl = 'http://localhost:4444';
             }
 
             $this->desiredCapabilities->setBrowserName($browserName);
         }
 
-        if ($this->createWebDriver) {
-            $this->driver = RemoteWebDriver::create(
-                $this->serverUrl,
-                $this->desiredCapabilities,
-                $this->connectionTimeout,
-                $this->requestTimeout,
-                null,
-                null,
-                null
-            );
-        }
+        $this->createWebDriver();
     }
 
     protected function tearDown(): void
     {
-        if ($this->driver instanceof RemoteWebDriver && $this->driver->getCommandExecutor()) {
+        if ($this->driver) {
```

Comment:
We should compare `$this->driver` with something, instead of just loose truthy comparison...

Reply:
added `!== null`